portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,35 +13,77 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
 import java.util.ArrayList;
 import java.util.List;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     private ArrayList<String> comments;
+    public void init(){
+        comments = new ArrayList<>();
+        
+    }
 
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-    comments.add("You're so smart");
-    comments.add("Keep up the good work");
-    comments.add(" Nice job! ");
-  }
-    
-    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("application/json;");
+  
+
+    // convert to json
     Gson gson = new Gson();
     String json = gson.toJson(comments);
     response.getWriter().println(json);
-  
+    
 }
+ @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "userName", "");
+    String comment = getParameter(request, "comment", "");
+   // System.out.println(comment);
+   for(String x : comments){
+       System.out.println("comment box in dataservlet "+x);
+       System.out.println("new line");
+   }
+    // Respond with the result.
+    response.setContentType("text/html;");
+    comments.add(comment+"            --"+name);
+      Entity comEntity = new Entity("Task");
+    comEntity.setProperty("name", name);
+    comEntity.setProperty("comment", comment);
+
+    //using the data store to make comments persist
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(comEntity);
```

Comment:
line 65-71: It looks like you created the datastore here and added entries to it, but never read the entries (the reading would be in doGet in DataServlet). Either add the part where you read from data store, or delete this data store part in doPost. If you do want to implement the rest of the data store function, I recommend using an Entity name that is more relevant to your project, eg Entity("Post") instead of "Task"